crates/ere-dockerized/src/lib.rs

```diff
@@ -0,0 +1,451 @@
+#![cfg_attr(not(test), warn(unused_crate_dependencies))]
+
+use crate::{
+    docker::{DockerBuildCmd, DockerRunCmd, docker_image_exists},
+    error::{CommonError, CompileError, DockerizedError, ExecuteError, ProveError, VerifyError},
+};
+use cargo_metadata::MetadataCommand;
+use serde::{Deserialize, Serialize};
+use std::{
+    env,
+    fmt::{self, Display, Formatter},
+    fs,
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+use tempfile::TempDir;
+use zkvm_interface::{
+    Compiler, Input, ProgramExecutionReport, ProgramProvingReport, ProverResourceType, zkVM,
+    zkVMError,
+};
+
+include!(concat!(env!("OUT_DIR"), "/crate_version.rs"));
+include!(concat!(env!("OUT_DIR"), "/zkvm_sdk_version_impl.rs"));
+
+pub mod docker;
+pub mod error;
+pub mod input;
+
+#[derive(Clone, Copy, Debug)]
+pub enum ErezkVM {
+    Jolt,
+    Nexus,
+    OpenVM,
+    Pico,
+    Risc0,
+    SP1,
+    Zisk,
+}
+
+impl ErezkVM {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            Self::Jolt => "jolt",
+            Self::Nexus => "nexus",
+            Self::OpenVM => "openvm",
+            Self::Pico => "pico",
+            Self::Risc0 => "risc0",
+            Self::SP1 => "sp1",
+            Self::Zisk => "zisk",
+        }
+    }
+
+    pub fn base_tag(&self, version: &str) -> String {
+        format!("ere-base:{version}")
+    }
+
+    pub fn base_zkvm_tag(&self, version: &str) -> String {
+        format!("ere-base-{self}:{version}")
+    }
+
+    pub fn cli_zkvm_tag(&self, version: &str) -> String {
+        format!("ere-cli-{self}:{version}")
+    }
+
+    pub fn build_docker_image(&self) -> Result<(), CommonError> {
+        let workspace_dir = workspace_dir();
+
+        let force_rebuild = env::var("ERE_FORCE_REBUILD_DOCKER_IMAGE")
+            .ok()
+            .and_then(|s| s.parse::<bool>().ok())
+            .unwrap_or_default();
```

Comment:
Re https://github.com/eth-act/ere/pull/66#discussion_r2242784471 indeed force rebuild seems safer.

Reply:
I think the safer option is the one you did initially :smile::
- Original take: If no env is provided, it will always rebuild
- New take: If not env is provided, it won't rebuild (i.e., use cached).

I'm fine with both -- just double checking here if you really intended to do the safer one?